Board.cpp

```diff
@@ -1,189 +1,277 @@
+	if (n != -1 && squares[n].color == c && squares[n].ptype == pawn)
+		return true;
+	n = mailbox[mailbox64[square] + m * 11];
+	if (n != -1 && squares[n].color == c && squares[n].ptype == pawn)
+		return true;
+	return false;
+}
 
-		bool slide[6] = {false, false, true, true, true, false};
-		int offsets[6] = {0, 8, 4, 4, 8, 8};
-		int offset[6][8] = {
-			{   0,   0,  0,  0, 0,  0,  0,  0 },
-			{ -21, -19,-12, -8, 8, 12, 19, 21 },
-			{ -11,  -9,  9, 11, 0,  0,  0,  0 },
-			{ -10,  -1,  1, 10, 0,  0,  0,  0 },
-			{ -11, -10, -9, -1, 1,  9, 10, 11 },
-			{ -11, -10, -9, -1, 1,  9, 10, 11 }
-		};
-
-		Piece p;
-		for (uint8_t i = 0; i < 64; ++i)
+set<int> Board::genMoves() const
+{
+	set<int> moves;
+	Move move;
+
+	Piece p;
+	for (uint8_t i = 0; i < 64; ++i)
+	{
+		if (squares[i].color == side)
 		{
-			if (squares[i].color == side)
+			p = squares[i];
+			if (p.ptype != pawn)
 			{
-				p = squares[i];
-				if (p.ptype != pawn)
+				for (int j = 0; j < offsets[p.ptype - pawn]; ++j)
 				{
-					for (int j = 0; j < offsets[p.ptype - pawn]; ++j)
+					for (int n = i;;)
 					{
-						for (int n = i;;)
+						n = mailbox[mailbox64[n] + offset[p.ptype - pawn][j]];
+						if (n == -1) break;
+						if (squares[n].color != none)
 						{
-							n = mailbox[mailbox64[n] + offset[p.ptype - pawn][j]];
-							if (n == -1) break;
-							if (squares[n].color != none)
+							if (squares[n].color == xside)
 							{
-								if (squares[n].color == xside)
-								{
-									// capture
-									move.m = {64, i, static_cast<uint8_t>(n)};
-									moves.insert(move.x);
-								}
-								break;
+								// capture
+								move.m = {64, i, static_cast<uint8_t>(n)};
+								moves.insert(move.x);
 							}
-							move.m = {0, i, static_cast<uint8_t>(n)};
-							moves.insert(move.x);
-							if (!slide[p.ptype - pawn]) break;
+							break;
 						}
+						move.m = {0, i, static_cast<uint8_t>(n)};
+						moves.insert(move.x);
+						if (!slide[p.ptype - pawn]) break;
 					}
 				}
-				else // pawn moves
+			}
+			else // pawn moves
+			{
+				int n, m = (side == white) ? 1 : -1;
+				if (squares[i + m * 8].color == none)
 				{
-					int n, m = (side == white) ? 1 : -1;
-					if (squares[i + m * 8].color == none)
-					{
-						move.m = {0, i, static_cast<uint8_t>(i + m * 8)};
-						moves.insert(move.x);
-						if (side == white && (i / 8 == 1) && squares[i + m * 16].color == none)
-						{
-							move.m = {0, i, static_cast<uint8_t>(i + m * 16)};
-							moves.insert(move.x);
-						}
-						if (side == black && (i / 8 == 6) && squares[i + m * 16].color == none)
-						{
-							move.m = {0, i, static_cast<uint8_t>(i + m * 16)};
-							moves.insert(move.x);
-						}
-					}
-					n = mailbox[mailbox64[i] + m * -9];
-					if (n != -1 && squares[n].color == xside)
+					move.m = {0, i, static_cast<uint8_t>(i + m * 8)};
+					moves.insert(move.x);
+					if (side == white && (i / 8 == 1) && squares[i + m * 16].color == none)
```

Comment:
i/8 ==1. I'm confused. If you're checking if it is equal to 8 why not i==8.  if cheking if it is divisible, i % 8 ==0.

Reply:
or are you checking if i is 8 through 16 cause of int casting?

Reply:
Yea since it rounds down this is to check if a square is on the second rank. i/ 8 == 6 checks if you are on the 7th rank

Reply:
gotcha gotcha. thanks!